dali/operators/image/paste/multipaste.cc

```diff
@@ -0,0 +1,159 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/paste/multipaste.h"
+#include "dali/kernels/imgproc/paste/paste.h"
+#include "dali/core/tensor_view.h"
+
+namespace dali {
+
+DALI_SCHEMA(MultiPaste)
+.DocStr(R"code(Performs multiple pastes from image batch to each of outputs
+
+This operator can also change the type of data.)code")
+.NumInput(1)
+.InputDox(0, "images", "3D TensorList", R"code(Batch of input images.
+
+Assumes HWC layout.)code")
+.AddArg("in_ids", R"code(1D TensorList of type int.
+
+Indexes from what inputs to paste data in each iteration.)code", DALI_INT32, true)
+.AddOptionalArg<int>("in_anchors", R"code(2D TensorList of type int64
+
+Absolute values of LU corner of the selection for each iteration.
+Zeros are used if this is omitted.)code", nullptr, true)
+.AddOptionalArg<int>("shapes", R"code(2D TensorList of type int64
+
+Absolute values of size of the selection for each iteration.
+Input size is used if this is omitted.)code", nullptr, true)
+.AddOptionalArg<int>("out_anchors", R"code(2D TensorList of type int64
+
+Absolute values of LU corner of the paste for each iteration.
+Zeros are used if omitted.)code", nullptr, true)
+.AddArg("output_size",
+R"code(Output size.)code", DALI_INT_VEC, true)
+.AddOptionalArg("dtype",
+R"code(Output data type. If not set, the input type is used.)code", DALI_NO_TYPE)
+.NumOutput(1);
+
+DALI_REGISTER_OPERATOR(MultiPaste, MultiPasteCPU, CPU)
+
+bool MultiPasteCPU::SetupImpl(std::vector<OutputDesc> &output_desc,
+                              const workspace_t<CPUBackend> &ws) {
+  AcquireArguments(spec_, ws);
+
+  const auto &images = ws.template InputRef<CPUBackend>(0);
+  const auto &output = ws.template OutputRef<CPUBackend>(0);
+  output_desc.resize(1);
+
+  TYPE_SWITCH(images.type().id(), type2id, InputType, (uint8_t, int16_t, int32_t, float), (
+      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float), (
+          {
+            using Kernel = kernels::PasteCPU<OutputType, InputType>;
+            kernel_manager_.Initialize<Kernel>();
+            TensorListShape<> sh = images.shape();
+            TensorListShape<> shapes(sh.num_samples(), sh.sample_dim());
+            for (int i = 0; i < sh.num_samples(); i++) {
+              const TensorShape<> &out_sh = { output_size_[i].data[0],
+                                              output_size_[i].data[1], sh[i][2] };
+              shapes.set_tensor_shape(i, out_sh);
+            }
+
+            output_desc[0] = {shapes, TypeTable::GetTypeInfo(output_type_)};
+          }
+      ), DALI_FAIL(make_string("Unsupported output type: ", output_type_)))  // NOLINT
+  ), DALI_FAIL(make_string("Unsupported input type: ", images.type().id())))  // NOLINT
+  return true;
+}
+
+
+template<typename InputType, typename OutputType>
+void MultiPasteCPU::RunImplExplicitlyTyped(workspace_t<CPUBackend> &ws) {
```

Comment:
nitpick: The name is a bit too long. Perhaps `RunImplExplicitlyTyped`?
(You can disregard this suggestion)